numpy/core/_methods.py

```diff
@@ -63,7 +63,10 @@ def _count_reduce_items(arr, axis):
         axis = (axis,)
     items = 1
     for ax in axis:
-        items *= arr.shape[ax]
+        try:
+            items *= arr.shape[ax]
+        except IndexError:
+            raise _exceptions.AxisError(ax, arr.ndim) from None
```

Comment:
This approach is a little unusual, but it does look correct, and is probably faster than calling `normalize_axis_index`.

I don't really mind either way, but if we keep it as you've written, there should be a comment indications you deliberately did not use ``normalize_axis_index`.

Reply:
Another thing to consider - if `normalize_axis_index` is not used, you can get `AxisError`s with negative axes values, e.g.:
```python
>>> np.arange(10).mean(axis=-2)
AxisError: axis -2 is out of bounds for array of dimension 1
```
I don't know if that's consistent with behavior elsewhere in NumPy or whether it's something that should be considered - just something that I noticed.

Reply:
Thank you for reviewing! I had no other reason than not being aware of `normalize_axis_index`. Using that is probably more consistent with the rest of the codebase, so I will update my PR.